writer.go

```diff
@@ -46,12 +53,21 @@ func (mw *messageWriter) Write(p []byte) (int, error) {
 			l = append(l, '\n')
 		}
 
-		_, err := mw.writeLine(l)
+		_, err = mw.writeLine(l)
 		N += n
 		if err != nil {
-			return N, err
+			break
```

Comment:
Note, ideally we should take the returned number of bytes by `writeLine` into account, instead of assuming everything has been written. This is made trickier by the fact we do some CRLF replacement above. Maybe something like:

```go
n, err = mw.writeLine(l)
N += n
if n == len(l) && replCRLF {
  N++
}
```

Reply:
Yeah, the CRLF replacement is a pain in itself, because as far as I could see, if we return less than len(p), the IO layer will complain that it's a short write. I was planning to look into this in another PR.